test/go.mod

```diff
@@ -17,6 +17,33 @@ require (
 	golang.org/x/sync v0.0.0-20201020160332-67f06af15bc9
 	google.golang.org/grpc v1.30.0
 	k8s.io/cri-api v0.20.1
+	k8s.io/kubernetes v1.20.4
 )
 
-replace github.com/Microsoft/hcsshim => ../
+replace (
+	github.com/Microsoft/hcsshim => ../
+	k8s.io/api => k8s.io/kubernetes/staging/src/k8s.io/api v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/apiextensions-apiserver => k8s.io/kubernetes/staging/src/k8s.io/apiextensions-apiserver v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/apimachinery => k8s.io/kubernetes/staging/src/k8s.io/apimachinery v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/apiserver => k8s.io/kubernetes/staging/src/k8s.io/apiserver v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/cli-runtime => k8s.io/kubernetes/staging/src/k8s.io/cli-runtime v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/client-go => k8s.io/kubernetes/staging/src/k8s.io/client-go v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/cloud-provider => k8s.io/kubernetes/staging/src/k8s.io/cloud-provider v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/cluster-bootstrap => k8s.io/kubernetes/staging/src/k8s.io/cluster-bootstrap v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/code-generator => k8s.io/kubernetes/staging/src/k8s.io/code-generator v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/component-base => k8s.io/kubernetes/staging/src/k8s.io/component-base v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/component-helpers => k8s.io/kubernetes/staging/src/k8s.io/component-helpers v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/controller-manager => k8s.io/kubernetes/staging/src/k8s.io/controller-manager v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/cri-api => k8s.io/kubernetes/staging/src/k8s.io/cri-api v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/csi-translation-lib => k8s.io/kubernetes/staging/src/k8s.io/csi-translation-lib v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kube-aggregator => k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kube-controller-manager => k8s.io/kubernetes/staging/src/k8s.io/kube-controller-manager v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kube-proxy => k8s.io/kubernetes/staging/src/k8s.io/kube-proxy v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kube-scheduler => k8s.io/kubernetes/staging/src/k8s.io/kube-scheduler v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kubectl => k8s.io/kubernetes/staging/src/k8s.io/kubectl v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/kubelet => k8s.io/kubernetes/staging/src/k8s.io/kubelet v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/legacy-cloud-providers => k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/metrics => k8s.io/kubernetes/staging/src/k8s.io/metrics v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/mount-utils => k8s.io/kubernetes/staging/src/k8s.io/mount-utils v0.0.0-20210218160201-e87da0bd6e03
+	k8s.io/sample-apiserver => k8s.io/kubernetes/staging/src/k8s.io/sample-apiserver v0.0.0-20210218160201-e87da0bd6e03
```

Comment:
can you explain why this is needed?

Reply:
Basically, because k8s has the same thing in their go.mod ([link](https://github.com/kubernetes/kubernetes/blob/master/go.mod#L143)). `replace` directives are not "inherited" by others who import your module, so if we don't have our own `replace` directives, we try to use the version of these modules that are in the `require` section in k8s's go.mod, which is `v0.0.0`.

As an example, see the entry for [`k8s.io/cri-api`](https://github.com/kubernetes/kubernetes/blob/master/go.mod#L143) in k8s go.mod, set to `v0.0.0`, and then `replace`d [here](https://github.com/kubernetes/kubernetes/blob/master/go.mod#L506).